src/cli/commands.rs

```diff
@@ -0,0 +1,94 @@
+use sc_cli::CliConfiguration;
+use sc_service::Configuration;
+use std::fmt::Debug;
+use std::{fs, io, path};
+use structopt::StructOpt;
+
+pub use pallet_abci;
+
+/// The `purge-abci-storage` command used to remove abci storage.
+#[derive(Debug, StructOpt)]
+pub struct PurgeChainWithStorageCmd {
+    #[allow(missing_docs)]
+    #[structopt(flatten)]
+    pub shared_params: sc_cli::SharedParams,
+    #[structopt(short = "y")]
+    pub yes: bool,
+}
+
+fn remove_rocks_db(db_name: &str, db_path: path::PathBuf) {
+    match fs::remove_dir_all(&db_path) {
+        Ok(_) => {
+            println!("{:?} removed.", &db_name);
+        }
+        Err(_) => {
+            println!("{:?} did not exist.", &db_path);
+        }
+    }
+}
+
+impl PurgeChainWithStorageCmd {
+    /// Run the purge command.
+    pub fn run(&self, config: &Configuration) -> sc_cli::Result<()> {
+        let chain_spec_id = config.chain_spec.id();
+        let shared_params = self.shared_params();
+        let database_params = self.database_params().unwrap_or(&sc_cli::DatabaseParams {
+            database: None,
+            database_cache_size: None,
+        });
+        let config_dir = config
+            .base_path
+            .as_ref()
+            .ok_or_else(|| "base_path has not been set")
+            .unwrap()
+            .path()
+            .to_path_buf()
+            .join("chains")
+            .join(chain_spec_id);
+        let rocks_db_name = &pallet_abci::get_storage_name();
+        let db_path = config_dir.join(&rocks_db_name);
+
+        if !self.yes {
+            println!("Are you sure to remove {:?}? [y/N]: ", &db_path);
```

Comment:
Ask that you want to remove all databases not only rocks db. If answer is 'yes' remove both database, the same with the non-interactive mode. 
As a result application should not ask user TWICE to reset the state.